src/topology/mod.rs

```diff
@@ -79,11 +79,28 @@ pub async fn start_validated(
 ) -> Option<(RunningTopology, mpsc::UnboundedReceiver<()>)> {
     let (abort_tx, abort_rx) = mpsc::unbounded_channel();
 
+    let expire_metrics = match (
+        config.global.expire_metrics,
+        config.global.expire_metrics_secs,
+    ) {
+        (e @ Some(_), None) => {
+            warn!(
+                "Global `expire_metrics` setting is deprecated and will be removed in a future version. Use `expire_metrics_secs` instead."
```

Comment:
Could we add this deprecation to the 0.25.0 upgrade guide too?

Big 👍 on Vector logging this though.

In https://github.com/vectordotdev/vector/blob/master/docs/DEPRECATION.md#deprecation we say that any deprecation messages logged by Vector should lead with `DEPRECATED` to make them easy to spot / grep for.